src/main/java/com/phonebookservice/controller/AbstractController.java

```diff
@@ -0,0 +1,33 @@
+package main.java.com.phonebookservice.controller;
+
+import main.java.com.phonebookservice.model.Model;
+import main.java.com.phonebookservice.server.DBIf;
+
+public abstract class AbstractController {
+    private DBIf databse = null;
+
+    /**
+     * controller constructor.
+     *
+     * @param databseValue the database
+     */
+    public AbstractController(final DBIf databseValue) {
```

Comment:
Constructor in abstract class should be `protected` 

You should answer this question :- Why should be `protected` ?

Reply:
as only any other child classes or classes within the same package can instantiate this class.

Reply:
Wrong answer because this class is protected, nobody can initiate this AbstractClass because it is abstract. 

Try to give another answer

Reply:
Since abstract can not be initiated, so only the subclass can access the constructor so it does not make sense to make it public and any other class can not instantiate it.